dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/response/SolrSearchResult.java

```diff
@@ -909,7 +857,7 @@ public String getDatasetUrl() {
              */
             String badString = "null";
             if (!identifier.contains(badString)) {
-                if (entity != null && entity instanceof Dataset) {
+                if (entity != null && type == SearchObjectType.DATASETS) {
                     if (this.isHarvested() && ((Dataset) entity).getHarvestedFrom() != null) {
```

Comment:
This one is kind of tricky.
We can see that `getDatasetUrl` is used in `search-include-fragment` and `mydata_fragment.xhtml`. But in such cases the `entity` comes from `public void populateDatasetSearchCard(Collection<SolrSearchResult> solrResults) {` in `SolrSearchResultsService` which will not fill `HarvestingClient` in `Dataset`. So `((Dataset) entity).getHarvestedFrom()` will always be null.